### PR TITLE
Added missing redirect abilities Gen 3-4 config

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Adds missing Lightning Rod/Storm Drain config for their Gen 3-4 behavior (redirect but don't grant immunity or stat boost).
Tested that tests account for the config.

Also:
- Adds missing AI checks for Motor Drive, Water Absorb, Dry Skin and Earth Eater as a follow up to #4781
- Fixed Ion Deluge's test.
- Consolidated Callnatives that use `IsElectricAbilityAffected`.
- Simplified `FindMonThatAbsorbsOpponentsMove` so it's easier to add new abilities.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -16350,7 +16350,7 @@ void BS_JumpIfEmergencyExited(void)
 void BS_JumpIfRod(void)
 {
     NATIVE_ARGS(const u8 *jumpInstr);
-    if (IsElectricAbilityAffected(ABILITY_LIGHTNING_ROD))
+    if (B_REDIRECT_ABILITY_IMMUNITY >= GEN_6 && IsElectricAbilityAffected(ABILITY_LIGHTNING_ROD))
```
Comment (Sneed69) [inline src/battle_script_commands.c:16353]: Is this intentionally 6 or should it be 5?

Comment (AsparagusEduardo) [reply src/battle_script_commands.c:16353]: Should be 5, thanks for the catch :)

**Comment by AsparagusEduardo** [comment]
Ready for re-review + updated description

**Comment by AlexOn1ine** [comment]
All of them are already handled in [IsDamageMoveUnusable](https://github.com/rh-hideout/pokeemerald-expansion/blob/upcoming/src/battle_ai_util.c#L421). The only 3 that need specific handling are motor drive, lightning rod, volt abosrb because of thunder wave and flash fire because of will o wisp.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**